### DISCUSSION TITLE
cannot decode PEM encoded block

### DISCUSSION THREAD

**Discussion by Sharkflyee** [qdm12/gluetun#1692] [Q&A]
ERROR reading from files: OpenVPN: client key: extracting base64 encoded data from PEM content: cannot decode PEM encoded block

2023-06-16T00:34:41Z INFO Shutdown successful

I copied private key from .ovpn file for vpn unlimited but keep on getting this error

**Comment by qdm12** [comment]
Make sure you follow steps from https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/openvpn-client-key.md

You are likely missing the header and footer like `-----BEGIN PRIVATE KEY-----`

**Reply by TheoOliveira** [reply]
what is it when my docker composer is set to use wireguard but keeps giving error that it's open vpn and can't open the pem file?
